pkg/resources/secret_with_oauth_client_credentials.go

```diff
@@ -0,0 +1,169 @@
+		},
+		"oauth_scopes": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Required:    true,
+			Description: "Specifies a list of scopes to use when making a request from the OAuth server by a role with USAGE on the integration during the OAuth client credentials flow.",
+		},
+	}
+	return helpers.MergeMaps(secretCommonSchema, secretClientCredentials)
+}()
+
+func SecretWithClientCredentials() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: CreateContextSecretWithClientCredentials,
+		ReadContext:   ReadContextSecretWithClientCredentials,
+		UpdateContext: UpdateContextSecretWithClientCredentials,
+		DeleteContext: DeleteContextSecretWithClientCredentials,
+
+		Schema: secretClientCredentialsSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "Secret with OAuth Client Credentials where Secrets Type attribute is set to OAUTH2.",
+	}
+}
+
+func CreateContextSecretWithClientCredentials(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	commonCreate := handleSecretCreate(d)
+
+	apiIntegrationString := d.Get("api_authentication").(string)
+	apiIntegration, err := sdk.ParseAccountObjectIdentifier(apiIntegrationString)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := sdk.NewSchemaObjectIdentifier(commonCreate.database, commonCreate.schema, commonCreate.name)
+
+	stringScopes := expandStringList(d.Get("oauth_scopes").(*schema.Set).List())
+	oauthScopes := make([]sdk.ApiIntegrationScope, len(stringScopes))
+	for i, scope := range stringScopes {
+		oauthScopes[i] = sdk.ApiIntegrationScope{Scope: scope}
+	}
+
+	request := sdk.NewCreateWithOAuthClientCredentialsFlowSecretRequest(id, apiIntegration).WithOauthScopes(sdk.OauthScopesListRequest{OauthScopesList: oauthScopes})
+
+	if commonCreate.comment != nil {
+		request.WithComment(*commonCreate.comment)
+	}
+
+	err = client.Secrets.CreateWithOAuthClientCredentialsFlow(ctx, request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeResourceIdentifier(id))
+
+	return ReadContextSecretWithClientCredentials(ctx, d, meta)
+}
+
+func ReadContextSecretWithClientCredentials(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	secret, err := client.Secrets.ShowByID(ctx, id)
+	if secret == nil || err != nil {
+		if errors.Is(err, sdk.ErrObjectNotFound) {
+			d.SetId("")
+			return diag.Diagnostics{
+				diag.Diagnostic{
+					Severity: diag.Warning,
+					Summary:  "Failed to retrieve secret. Target object not found. Marking the resource as removed.",
```

Comment:
It's also to give details here like in the diag below (it's easier to detect for which id this error happened; for us and the user).

Reply:
Added the details with fullyQualifiedName